### PR TITLE
Follow #337: use actionlib API in BufferClient::processGoal()

### PR DESCRIPTION
**Pull request by MichaelGrupp**
Follows the changes to the Python API from #337 to simplify the action client logic.

### CODE REVIEW

**Comment by MichaelGrupp** [comment]
The second commit probably needs to be changed, depending on how you want to handle parameter deprecation.

File: tf2_ros/include/tf2_ros/buffer_client.h [hunk 1]
```diff
@@ -132,7 +132,6 @@ namespace tf2_ros
       geometry_msgs::TransformStamped processGoal(const tf2_msgs::LookupTransformGoal& goal) const;
       geometry_msgs::TransformStamped processResult(const tf2_msgs::LookupTransformResult& result) const;
       mutable LookupActionClient client_;
-      double check_frequency_;
```
Comment (tfoote) [inline tf2_ros/include/tf2_ros/buffer_client.h]: If we leave this here with a note that it's unused then this change can be ABI compatible and can then be made available in active distros without need for a discussion.

File: tf2_ros/src/buffer_client.cpp [hunk 2]
```diff
@@ -40,7 +40,6 @@ namespace tf2_ros
 {
   BufferClient::BufferClient(std::string ns, double check_frequency, ros::Duration timeout_padding): 
     client_(ns), 
-    check_frequency_(check_frequency),
```
Comment (tfoote) [inline tf2_ros/src/buffer_client.cpp]: If we keep the member for compatibility we should still initialize it to avoid warnings.

**Review by tfoote — CHANGES_REQUESTED** [review]
Thanks this is good to do as well. You're right about the 2nd commit. It's good to clean up, but stability is more valuable for now than saving the memory of a single double value.

**Comment by MichaelGrupp** [comment]
Thanks for the quick review. I changed the 2nd commit to be ABI compliant.

**Review by tfoote — APPROVED** [review]

**Comment by tfoote** [comment]
Thank you for the patch and iterating on it!

**Pull request merged**